url64b10c0106_schlabberdog_blocks_tgz-pJ8-com_github_users_schlabberdog_blocks_ui_CliUIJ8/src/com/github/users/schlabberdog/blocks/r010/LUBlock.java

```diff
@@ -0,0 +1,125 @@
+package com.github.users.schlabberdog.blocks.r010;
+
+import com.github.users.schlabberdog.blocks.board.Block;
+import com.github.users.schlabberdog.blocks.board.Board;
+import com.github.users.schlabberdog.blocks.board.moves.IMove;
+import com.github.users.schlabberdog.blocks.board.moves.Move;
+import com.github.users.schlabberdog.blocks.board.moves.MultiMove;
+import com.github.users.schlabberdog.blocks.mccs.Coord;
+import com.github.users.schlabberdog.blocks.mccs.Rect;
+
+import java.awt.*;
+import java.util.ArrayList;
+
+/*  |XX
+ *  |X
+ *  |XX
+ */
+public class LUBlock extends Block {
+
+    public LUBlock() {
+        super(2,3);
+    }
+
+    @Override
+    public void printOntoMap(Coord pos,char[][] m) {
+        char c = getRepresentation();
+        m[pos.y][pos.x] = c;
+        m[pos.y][pos.x+1] = c;
+        m[pos.y+1][pos.x] = c;
+        m[pos.y+2][pos.x] = c;
+        m[pos.y+2][pos.x+1] = c;
+    }
+
+    @Override
+    public Color getColor() {
+        return Color.magenta;
+    }
+
+    @Override
+    public String toString() {
+        return "[LU]";
+    }
+
+    @Override
+    public Rect[] getRectSet(Coord pos) {
+        return new Rect[]{
+                new Rect(pos.x,pos.y  ,2,1),
+                new Rect(pos.x,pos.y  ,1,3),
+                new Rect(pos.x,pos.y+2,2,1)
+        };
+    }
+
+    @Override
+    public char getRepresentation() {
+        return 'C';
+    }
+
+    @Override
+    public void addAlts(Coord pos,Board board, ArrayList<IMove> alts) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 26. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.